Spigot-Server-Patches/0680-Remove-streams-from-SensorNearest.patch

```diff
@@ -0,0 +1,140 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Bjarne Koll <lynxplay101@gmail.com>
+Date: Wed, 3 Mar 2021 02:40:57 +0100
+Subject: [PATCH] Remove streams from SensorNearest
+
+The behavioural nearby sensors are validated every tick on the entities
+that registered the respective sensors and are therefore a good subject
+to performance improvements.
+
+More specifically this commit replaces the Stream#filter usage with
+ArrayList#removeIf as the removeIf method on an array list is heavily
+optimized towards a single internal array re-allocation without any
+further overhead on the removeIf call.
+
+The only negative of this change is the rather agressive diff these
+patches introduce as the methods are basically being reimplemented
+compared to the previous stream-based implementation.
+
+diff --git a/src/main/java/net/minecraft/server/SensorNearestItems.java b/src/main/java/net/minecraft/server/SensorNearestItems.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..c202f2101201242e074f28d1e85831e4c86bdea1
+--- /dev/null
++++ b/src/main/java/net/minecraft/server/SensorNearestItems.java
```

Comment:
Don't copy over nms files to access them. Read [this section](https://github.com/PaperMC/Paper/blob/master/CONTRIBUTING.md#i-cant-find-the-nms-file-i-need) of the contributing guide.

Reply:
Sorry maybe I understood the section wrong but I am clearly making changes to this file (see [here](https://github.com/PaperMC/Paper/pull/5295/files#diff-e6d430e057c14e1621bc4a179a5eab880460e3977525f53eaa8c35253ef3a452R43)). Doesn't that qualify for importing the class like this ?

Reply:
Import it as the section notes and do changes afterwards. We can't see what your changes are at the moment.

Reply:
I hope my recent push fixed the incorrectly imported file :)

Reply:
Tho I seem to struggle with the naming. Maybe I can reduce the diff

Reply:
One way you could reduce the diff is by commenting out the removed sections instead of removing them.

Reply:
I think complying with the previous naming conventions reduced it far enough (or I hope so). But yeah that would be a possibility